xls/public/c_api.cc

```diff
@@ -892,6 +916,30 @@ bool xls_package_get_function(struct xls_package* package,
   return false;
 }
 
+
+bool xls_package_get_functions(struct xls_package* package,
+                               char** error_out,
+                               struct xls_function*** result_out,
+                               size_t* count_out) {
+  CHECK(package != nullptr);
+  CHECK(error_out != nullptr);
+  CHECK(result_out != nullptr);
+  CHECK(count_out != nullptr);
+  xls::Package* xls_package = reinterpret_cast<xls::Package*>(package);
+  absl::Span<const std::unique_ptr<xls::Function>> functions = xls_package->functions();
+  *count_out = functions.size();
+  *result_out = new struct xls_function*[*count_out];
+
+  for (size_t i = 0; i < *count_out; ++i) {
```

Comment:
In some other cases (e.g. `ToOwnedCStrings`) I believe we avoid allocating the empty array and set the output to nullptr. This seems like another situation where that would make sense, WDYT?

Reply:
updated.